google_shopping.js

```diff
@@ -1,5 +1,90 @@
 var data = require("./products.json")
 
 // Write your solutions below
-
-console.log(data["items"]);
\ No newline at end of file
+console.log(data["items"]);
+console.log(data["title"]);
+var items = data.items;
+var product = data.title;
+/*
+1.) The kind of results you're are dealing are shopping#products. Go through the items and find all results that have kind of shopping#product. How many are there? Where else is this count information stored in the search results?
+*/
+var shopProd = [];
+for (var i = 0; i < (data["items"].length); i++){
+	if (data["items"][i]["kind"] === 'shopping#product') {
+		shopProd.push(i + " is a shopping product.");
+	}
+}
+console.log("---");
+console.log(shopProd.length + " items.");
+console.log("Check against 'Current Item Count': " + data["currentItemCount"]);
+if ((data["currentItemCount"]) === (shopProd.length)) {
+	console.log("Check is verfied!")
+	if (false) {
+		console.log("Check FAILED.");
+	}
+}
+console.log("---")
+/*
+2.) Find all items with a backorder availability in inventories.
+*/
+console.log("---");
+//var title = items.title
+var backOrder = [];
+for (var i = 0; i < (data["items"].length); i++) {
+	for (var b = 0; b < data["items"][i]["product"]["inventories"].length; b++) {
+		if ((data["items"][i]["product"]["inventories"][b]["availability"]) === "backorder") {
+		backOrder.push("Item " + [i] + " " + product + " is on backorder");
```

Comment:
Here, product returns undefined. Also, you don't need [] around i because they will be ignored. You can instead put the following:

```backOrder.push("Item " + i + " " + data["items"][i]["product"]["title"] + " is on backorder");```